kotangenss/rss-react
# React. Tests/Context API

1. Task: https://github.com/rolling-scopes-school/tasks/blob/master/react/modules/module03/README.md
2. Screenshot:
   
![screen](https://github.com/kotangenss/rss-react/assets/61937290/e12cf8af-7bf3-44f9-a290-fbf435b68fc4)

3. Deploy: https://654f12ec7dbdf36a2487e327--neon-smakager-85b41a.netlify.app/
4. Done 11.11.2023 / deadline 13.11.2023
5. Score: 100 / 100

<img src="https://media2.giphy.com/media/b73wUq1jFLBTy/giphy.gif"/>

# Necessary tests included in the assignment:

7. Tests for the Card component:
- Ensure that the card component renders the relevant card data;
Test - **«Сomponent renders the relevant card data»**
In **src/components/resultSection/index.test.tsx**

- Validate that clicking on a card opens a detailed card component;
Test - **«Clicking on a card opens a detailed card component»**
In **src/pages/main/index.test.tsx**

- Check that clicking triggers an additional API call to fetch detailed information.
Test - **«Clicking on a card opens a detailed card component»**
In **src/pages/main/index.test.tsx**

8. Tests for the Detailed Card component:
- Check that a loading indicator is displayed while fetching data;
Test - **«Renders Loader when isLoading is true»**
In **src/components/details/index.test.tsx**

- Make sure the detailed card component correctly displays the detailed card data;
Test - **«Renders details correctly»**
In **src/components/details/index.test.tsx**

- Ensure that clicking the close button hides the component.
Test - **«Close detailes component after close button is clicked»**
In **src/pages/main/index.test.tsx**

9. Tests for the Pagination component:
- Make sure the component updates URL query parameter when page changes.
Test - **«Successfully switches to next/prev pages»**
In **src/pages/main/index.test.tsx**

10. Tests for the Search component:
- Verify that clicking the Search button saves the entered value to the local storage;
Test - **«Run search with not empty input, successfully returns not empty items»**
In **src/components/searchInput/index.test.tsx**

-  Check that the component retrieves the value from the local storage upon mounting.
Test - **«Renders the input field and button with the provided props»**
In **src/components/searchInput/index.test.tsx**

11. Tests for the 404 Page component:
- Ensure that the 404 page is displayed when navigating to an invalid route.
Test - **«404 page appears when following an incorrect route»**
In **src/pages/notFound/index.test.tsx**

###
